### PR TITLE
Add bgp_device_global yang model

### PR DESCRIPTION
**Pull request by tjchadaga**
#### Why I did it
Needed a configDB table to hold global device-specific BGP configuration
This table was introduced as part of changes to make traffic-shift-away (TSA) state persistent - https://github.com/Azure/sonic-buildimage/pull/11257

#### How I did it
Add a new yang file sonic-bgp-device-global.yang and new test files

#### How to verify it
Compile target/python-wheels/sonic_yang_mgmt-1.0-py3-none-any.whl and target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

#### Link to config_db schema for YANG module changes

https://github.com/Azure/sonic-buildimage/blob/6fd7eab250ac54e9fedb68fc5248fdef2ba0adf6/src/sonic-yang-models/doc/Configuration.md#bgp-device-global

### CODE REVIEW

File: src/sonic-yang-models/tests/yang_model_tests/tests/bgp_device_global.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    "BGP_DEVICE_GLOBAL_WITH_TSA_TEST": {
+        "desc": "Load bgp device global table with tsa_enabled set to true"
+    },
+    "BGP_DEVICE_GLOBAL_WITH_TSB_TEST": {
+        "desc": "Load bgp device global table with tsa_enabled set to false"        
+    },
+    "BGP_DEVICE_GLOBAL_WITH_INVALID_VALUE": {
+        "desc": "Load bgp device global table with invalid value",
+        "eStrKey": "InvalidValue",
+	"eStr": ["tsa_enabled"]
```
Comment (qiluo-msft) [inline src/sonic-yang-models/tests/yang_model_tests/tests/bgp_device_global.json:11]: >	 [](http://example.com/codeflow?start=0&length=1)

Do not mix tabs with spaces. Prefer 4-space indentation.

**Comment by qiluo-msft** [comment]
Could you add HLD link in the PR description?

**Comment by tjchadaga** [comment]
> Could you add HLD link in the PR description?

PR link https://github.com/Azure/sonic-buildimage/pull/11257 included. This has the description and changes which brought in the new table

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/bgp_device_global.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                    "tsa_enabled": "true"
+                }
+            }
+        }
+    },
+    "BGP_DEVICE_GLOBAL_WITH_TSB_TEST": {
+        "sonic-bgp-device-global:sonic-bgp-device-global": {
+            "sonic-bgp-device-global:BGP_DEVICE_GLOBAL": {
+                "STATE":{
+                    "tsa_enabled": "false"
```
Comment (ganglyu) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/bgp_device_global.json:15]: Can you add a unit test for default value?

**Comment by zhangyanzhao** [comment]
Reviewed in YANG community

File: src/sonic-yang-models/doc/Configuration.md [hunk 3]
```diff
@@ -709,41 +723,41 @@ This kind of profiles will be handled by buffer manager and won't be applied to
...
-IP prefix as second-level key. IP interface address objects don't have any attributes.
-IP interface attributes, resides in those tables as well, key is the interface name
-and value is a list of field-values representing the interface attributes, e.g. loopback action.
+IP configuration for data plane are defined in **INTERFACE**, **VLAN_SUB_INTERFACE**,
+**PORTCHANNEL_INTERFACE** and **VLAN_INTERFACE** table. The objects
+in all four tables have the interface (could be physical port, port
+channel, vlan or vlan sub interface) that IP address is attached to as first-level key, and
+IP prefix as second-level key. IP interface address objects don't have any attributes.
+IP interface attributes, resides in those tables as well, key is the interface name
+and value is a list of field-values representing the interface attributes, e.g. loopback action.
```
Comment (praveen-li) [inline src/sonic-yang-models/doc/Configuration.md:732]: seems like white spaces diff due to IDE. Any way these are good to have.

**Review by praveen-li — APPROVED** [review]
Looks good to me. With Optional suggestions.

File: src/sonic-yang-models/setup.py [hunk 4]
```diff
@@ -81,6 +81,7 @@ def run(self):
                          './yang-models/sonic-auto_techsupport.yang',
                          './yang-models/sonic-bgp-common.yang',
                          './yang-models/sonic-bgp-global.yang',
+                         './yang-models/sonic-bgp-device-global.yang',
```
[the remark this answers is not in the document]
Comment (tjchadaga) [reply src/sonic-yang-models/setup.py:83]: Done

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/bgp_device_global.json (hunk 2 above)
Comment (tjchadaga) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/bgp_device_global.json:24]: Done

**Comment by qiluo-msft** [comment]
@ganglyu Any more concern?

**Review by ganglyu — APPROVED** [review]

**Comment by qiluo-msft** [comment]
This is not considered a bug fix. Why need it on 202012 branch?

**Pull request merged**